realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -437,7 +442,16 @@ private void completedAsyncQueriesUpdate(QueryUpdateTask.Result result) {
                 // refresh the Realm to the version provided by the worker thread
                 // (advanceRead to the latest version may cause a version mismatch error) preventing us
                 // from importing correctly the handover table view
-                realm.sharedGroupManager.advanceRead(result.versionID);
+                try {
+                    realm.sharedGroupManager.advanceRead(result.versionID);
+                } catch (BadVersionException e) {
```

Comment:
I think it is totally safe to continue using the `SharedGroup` even if the `advance_read` failed because of `BadVersion`, right? @finnschiermer